ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/InternalPaywallView.kt

```diff
@@ -20,50 +27,63 @@ import com.revenuecat.purchases.ui.revenuecatui.templates.Template3
 
 @Composable
 internal fun InternalPaywallView(
-    mode: PaywallViewMode = PaywallViewMode.default,
-    offering: Offering? = null,
-    listener: PaywallViewListener? = null,
-    viewModel: PaywallViewModel = getPaywallViewModel(offering = offering, listener = listener, mode = mode),
+    options: PaywallViewOptions,
+    viewModel: PaywallViewModel = getPaywallViewModel(options),
 ) {
     viewModel.refreshStateIfLocaleChanged()
     val colors = MaterialTheme.colorScheme
     viewModel.refreshStateIfColorsChanged(colors)
 
     when (val state = viewModel.state.collectAsState().value) {
         is PaywallViewState.Loading -> {
-            LoadingPaywallView(mode = mode)
+            LoadingPaywallView(mode = options.mode)
         }
         is PaywallViewState.Error -> {
             Text(text = "Error: ${state.errorMessage}")
         }
         is PaywallViewState.Loaded -> {
-            when (state.templateConfiguration.template) {
-                PaywallTemplate.TEMPLATE_1 -> Template1(state = state, viewModel = viewModel)
-                PaywallTemplate.TEMPLATE_2 -> Template2(state = state, viewModel = viewModel)
-                PaywallTemplate.TEMPLATE_3 -> Template3(state = state, viewModel = viewModel)
-                PaywallTemplate.TEMPLATE_4 -> Text(text = "Error: Template 4 not supported")
-                PaywallTemplate.TEMPLATE_5 -> Text(text = "Error: Template 5 not supported")
+            val backgroundColor = state.templateConfiguration.getCurrentColors().background
+            Box(
+                modifier = Modifier
+                    .conditional(state.isInFullScreenMode) {
+                        Modifier.background(backgroundColor)
+                    }
+                    .conditional(!state.isInFullScreenMode) {
+                        Modifier
+                            .clip(
+                                RoundedCornerShape(
+                                    topStart = footerRoundedBorderHeight,
+                                    topEnd = footerRoundedBorderHeight,
+                                ),
+                            )
+                            .background(backgroundColor)
+                            .padding(top = footerRoundedBorderHeight)
+                    },
+            ) {
+                when (state.templateConfiguration.template) {
+                    PaywallTemplate.TEMPLATE_1 -> Template1(state = state, viewModel = viewModel)
+                    PaywallTemplate.TEMPLATE_2 -> Template2(state = state, viewModel = viewModel)
+                    PaywallTemplate.TEMPLATE_3 -> Template3(state = state, viewModel = viewModel)
+                    PaywallTemplate.TEMPLATE_4 -> Text(text = "Error: Template 4 not supported")
+                    PaywallTemplate.TEMPLATE_5 -> Text(text = "Error: Template 5 not supported")
+                }
```

Comment:
This method is kinda busy now, maybe we can extract this part to a separate method now?